runtime-modules/content/src/types.rs

```diff
@@ -0,0 +1,560 @@
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub struct VideoCreationParameters<ContentParameters> {
+    /// Assets referenced by metadata
+    pub assets: Vec<NewAsset<ContentParameters>>,
+    /// Metadata for the video.
+    pub meta: Vec<u8>,
+}
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct VideoUpdateParameters<ContentParameters> {
+    /// Assets referenced by metadata
+    pub assets: Option<Vec<NewAsset<ContentParameters>>>,
+    /// If set, metadata update for the video.
+    pub new_meta: Option<Vec<u8>>,
+}
+
+/// A video which belongs to a channel. A video may be part of a series or playlist.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct VideoRecord<
+    ChannelId,
+    SeriesId,
+    BlockNumber: BaseArithmetic + Copy,
+    MemberId: Default + Copy + Ord,
+    Balance: Default,
+> {
+    pub in_channel: ChannelId,
+    // keep track of which season the video is in if it is an 'episode'
+    // - prevent removing a video if it is in a season (because order is important)
+    pub in_series: Option<SeriesId>,
+    /// Whether the curators have censored the video or not.
+    pub is_censored: bool,
+    /// Whether nft for this video have been issued.
+    pub nft_status: Option<OwnedNFT<BlockNumber, MemberId, Balance>>,
+}
+
+impl<
+        ChannelId: Clone,
+        SeriesId: Clone,
+        BlockNumber: BaseArithmetic + Copy,
+        MemberId: Default + Copy + PartialEq + Ord,
+        Balance: Clone + Default,
+    > VideoRecord<ChannelId, SeriesId, BlockNumber, MemberId, Balance>
+{
+    fn is_issued(&self) -> bool {
+        self.nft_status.is_some()
+    }
+
+    /// Ensure nft status is set to `NoneIssued`
+    pub fn ensure_nft_is_not_issued<T: Trait>(&self) -> DispatchResult {
+        ensure!(!self.is_issued(), Error::<T>::NFTAlreadyExists);
+        Ok(())
+    }
+
+    /// Ensure nft status is set to `Owned`
+    pub fn ensure_nft_is_issued<T: Trait>(&self) -> DispatchResult {
+        ensure!(self.is_issued(), Error::<T>::NFTDoesNotExist);
+        Ok(())
+    }
+
+    /// Ensure given NFTOwner is nft owner
+    pub fn ensure_nft_ownership<T: Trait>(&self, owner: &NFTOwner<MemberId>) -> DispatchResult {
+        if let Some(owned_nft) = &self.nft_status {
+            ensure!(owned_nft.is_owner(owner), Error::<T>::DoesNotOwnNFT);
+        }
+        Ok(())
+    }
+
+    /// Check whether nft transactional status is set to `Auction`
+    pub fn is_nft_auction_started(&self) -> bool {
+        matches!(
+            self.nft_status,
+            Some(OwnedNFT {
+                transactional_status: TransactionalStatus::Auction(..),
+                ..
+            })
+        )
+    }
+
+    /// Ensure nft is in auction state
+    pub fn ensure_nft_auction_state<T: Trait>(
```

Comment:
I don't agree with `ensure_has_active_auction` naming here. Because nft can be in auction state, but it is not active. (`starts_at`) flag condition is not satisfied. It's a completely different error!

Reply:
Please, look at `ensure_nft_auction_not_expired` & `ensure_auction_can_be_completed` functions

Reply:
Great.